internal/dynamo/client.go

```diff
@@ -90,6 +89,12 @@ func (c *Client) OneByUID(ctx context.Context, uid string, v interface{}) error
 		return NotFoundError{}
 	}
 
+	// limits are applied before filters so we need to manually handle > 1
+	// see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.Other.html#Query.Limit
+	if len(response.Items) > 1 {
+		return fmt.Errorf("expected to resolve partial PK and SK but got %d items", len(response.Items))
+	}
+
```

Comment:
I lost almost a full day to this - now we have scheduled events against LpaUID GSI if the donor key result isn't returned first then the limit is applied before the filter and we get no results.

e.g.

Won't work:

![Screenshot 2025-02-11 at 14 26 29](https://github.com/user-attachments/assets/a44302b7-f568-41f5-9a8a-1d0713bd4236)

Will work:

![Screenshot 2025-02-11 at 14 27 07](https://github.com/user-attachments/assets/c7517497-0c99-45a6-b0cb-a1fefa74c4dd)

I couldn't see a way to reliably sort to ensure the keys record is always first so this seemed like the next best thing.

Reply:
very fun, TIL

Reply:
I think I'd be happy without the error though, as it shouldn't happen. Or if we do want to error, include the UID so that we can figure out where the issue is